### ISSUE TITLE
user-deployments helm chart: expose Deployment.spec.revisionHistoryLimit

### ISSUE THREAD

**Issue by agon-shabi** [dagster-io/dagster#17433]
### What's the use case?

Looks like this defaults to 10 on kubernetes, and means we've got this many dead replicaset instances for each deployment at any time.

Not causing any material issues at the moment, but would be nice to be able to control this from the chart to reduce mess when looking at argo-cd (which is how we deploy).

### Ideas of implementation

Not fussed - happy for a single `revisionHistoryLimit` config value on the `UserDeployment` config type, which gets passed through to the rendered `Deployment`?

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by mzjp2** [comment]
+1 for this

**Comment by h4tt3n** [comment]
I agree that this would be very helpful. Our deployment is also flooded with dead replicasets that must be removed manually.

**Comment by mikou-mouad** [comment]
Guys thanks a lot for this, but i'm not sure on which chart version is it available?
I'm using the latest 1.11.1 but not seeing it.
